core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -389,9 +404,9 @@ private[spark] class MesosClusterScheduler(
         }
       val cmdExecutable = new File(executorSparkHome, "./bin/spark-submit").getCanonicalPath
       val cmdJar = desc.jarUrl.split("/").last
-      s"$cmdExecutable ${cmdOptions.mkString(" ")} $cmdJar $appArguments"
+      (cmdExecutable, cmdJar)
     }
-    builder.setValue(cmd)
+    builder.setValue(s"$executable ${cmdOptions.mkString(" ")} $jar $appArguments")
```

Comment:
in L373 up there can you just set `val cmdOptions = generateCmdOptions(desc).mkString(" ")`, so this is easier to read? I believe it's only used in one place so it's safe to do that.